core/src/main/java/org/ehcache/core/spi/store/Store.java

```diff
@@ -274,32 +277,29 @@
    * <p>
    * This is equivalent to
    * <pre>
-   *   V newValue = mappingFunction.apply(key, store.get(key));
+   *   V oldValue = store.get(key;
+   *   V newValue = mappingFunction.apply(key, oldValue);
    *   if (newValue != null) {
    *     store.put(key, newValue);
    *   } else {
    *     store.remove(key);
    *   }
-   *   return newValue;
+   *   return oldValue;
    * </pre>
    * except that the action is performed atomically.
-   * <p>
-   * This is equivalent to calling {@link Store#compute(Object, BiFunction, Supplier)}
-   * with a "replaceEquals" function that returns {@link Boolean#TRUE true}.
-   * <p>
    * Neither the key nor the function can be {@code null}
    *
    * @param key the key to update the mapping for
    * @param mappingFunction the function that will produce the new value.
-   * @return the new value associated with the key or {@code null} if none
+   * @return the existing value associated with the key or {@code null} if none
    *
    * @throws ClassCastException if the specified key is not of the correct type {@code K}
    * @throws NullPointerException if any of the arguments is {@code null}
    * @throws StoreAccessException if the mapping can't be changed
    *
-   * @see #compute(Object, BiFunction, Supplier)
+   * @see #compute(Object, BiFunction, Supplier, Supplier)
    */
-  ValueHolder<V> compute(K key, BiFunction<? super K, ? super V, ? extends V> mappingFunction) throws StoreAccessException;
+  ValueHolder<V> getAndCompute(K key, BiFunction<? super K, ? super V, ? extends V> mappingFunction) throws StoreAccessException;
```

Comment:
Given that you are adding `getAndCompute` we should probably not have general compute methods, but instead call those `computeAndGet` so that everything is clear.

Reply:
After I gave it a thought , isn't it like `put` and `getAndPut`, `remove` and `getAndRemove`, in that way the name does make sense to me

Reply:
It might make sense but I still think it's confusing.